tutorial/tracking/model-based/generic-aprilTag/tutorial-mb-generic-tracker-aprilTag-init.cpp

```diff
@@ -0,0 +1,325 @@
+    } else if (std::string(argv[i]) == "--help" || std::string(argv[i]) == "-h") {
+      std::cout << "Usage: " << argv[0] << " [--input <camera input>] [--tag_size <tag_size in m>]"
+                                           " [--quad_decimate <quad_decimate>] [--nthreads <nb>]"
+                                           " [--intrinsic <intrinsic file>] [--camera_name <camera name>]"
+                                           " [--pose_method <method> (0: HOMOGRAPHY, 1: "
+                                           "HOMOGRAPHY_VIRTUAL_VS,"
+                                           " 2: DEMENTHON_VIRTUAL_VS, 3: LAGRANGE_VIRTUAL_VS,"
+                                           " 4: BEST_RESIDUAL_VIRTUAL_VS)]"
+                                           " [--tag_family <family> (0: TAG_36h11, 1: TAG_36h10, 2: "
+                                           "TAG_36ARTOOLKIT,"
+                                           " 3: TAG_25h9, 4: TAG_25h7, 5: TAG_16h5)]";
+#if (defined(VISP_HAVE_X11) || defined(VISP_HAVE_GDI) || defined(VISP_HAVE_OPENCV))
+      std::cout << " [--display_off]";
+#endif
+      std::cout << " [--cube_edge_size <cube_edge_size in m>]" << std::endl;
+      std::cout << " [--texture]" << std::endl;
+      std::cout << " [--help]" << std::endl;
+      return EXIT_SUCCESS;
+    }
+  }
+
+  createCaoFile(cubeEdgeSize);
+
+  vpCameraParameters cam;
+  bool camIsInit = false;
+#ifdef VISP_HAVE_XML2
+  vpXmlParserCamera parser;
+  if (!intrinsic_file.empty() && !camera_name.empty()) {
+    parser.parse(cam, intrinsic_file, camera_name, vpCameraParameters::perspectiveProjWithoutDistortion);
+    camIsInit = true;
+  }
+#endif
+
+  try {
+    vpImage<unsigned char> I;
+
+//! [Construct grabber]
+#if defined(VISP_HAVE_V4L2)
+    vpV4l2Grabber g;
+    std::ostringstream device;
+    device << "/dev/video" << opt_device;
+    g.setDevice(device.str());
+    g.setScale(1);
+    g.acquire(I);
+#elif defined(VISP_HAVE_OPENCV)
+    cv::VideoCapture cap(opt_device); // open the default camera
+    if (!cap.isOpened()) {            // check if we succeeded
+      std::cout << "Failed to open the camera" << std::endl;
+      return EXIT_FAILURE;
+    }
+    cv::Mat frame;
+    cap >> frame; // get a new frame from camera
+    vpImageConvert::convert(frame, I);
+#endif
+    if (!camIsInit) {
+      cam.initPersProjWithoutDistortion(600, 600, I.getWidth() / 2., I.getHeight() / 2.);
+    }
+
+    std::cout << "cam:\n" << cam << std::endl;
+    std::cout << "poseEstimationMethod: " << poseEstimationMethod << std::endl;
+    std::cout << "tagFamily: " << tagFamily << std::endl;
+
+    // Construct display
+    vpDisplay *d = NULL;
+    if (!display_off) {
+#ifdef VISP_HAVE_X11
+      d = new vpDisplayX(I);
+#elif defined(VISP_HAVE_GDI)
+      d = new vpDisplayGDI(I);
+#elif defined(VISP_HAVE_OPENCV)
+      d = new vpDisplayOpenCV(I);
+#endif
+    }
+
+    // Prepare MBT
+    vpMbGenericTracker *tracker = new vpMbGenericTracker;
```

Comment:
I would advice to not use pointer in C++ when it can be avoided (as it seems to be here after a quick glance).